src/xdist/scheduler/load.py

```diff
@@ -201,7 +201,11 @@ def remove_node(self, node):
             return
 
         # The node crashed, reassing pending items
-        crashitem = self.collection[pending.pop(0)]
+        crashitem = self.collection[pending[0]]
+        if not self.config.hook.pytest_handlecrashitem(item=crashitem):
+            pending.pop(0)  # If false, item has not been handled, so remove item
```

Comment:
Always pop, the hook can put it back in

This hook may nerd company to manage detail like max worker restart

Im still not sure if its a good idea to just restart those crashers, they are tests that anhillated a python process, so it's the kind of flaky ness that requires different attention

Reply:
One reason for the crash is that a timeout occurred. In my case, I combine pytest-timeout with rerunfailures to rerun tests that timed out. However, pytest-rerunfailures is unable to catch this because it runs inside the worker.

Reply:
What do you propose? AFAIK, this has to crash the worker, because there's no other way to stop running python code other than to kill it.

Reply:
the timeout plugin module has multiple ways to enable this, but i'll have to read into the details